nogotofail/mitm/connection/handlers/data/ssl.py

```diff
@@ -172,3 +175,66 @@ def on_ssl(self, client_hello):
                 self.log(logging.ERROR,
                         "Client enabled SSLv3 protocol without TLS_FALLBACK_SCSV")
             self.log_attack_event(data="SSLv3")
+
+
+@handler.passive(handlers)
+class NoForwardSecrecy(DataHandler):
+    name = "noforwardsecrecy"
+    description = (
+        "Detects selected server cipher suites which don't support "
+        "Diffie-Hellman key exchange (DHE or ECDHE) i.e. in "
+        "SERVER_HELLO response")
+
+    buffer = ""
+    ssl = False
+
+    def on_ssl(self, client_hello):
+        self.ssl = True
+        self.client_session_id = client_hello.session_id
+        return False
+
+    def on_response(self, response):
```

Comment:
(Sorry for the late response, holidays are busy :().

This would probably be easier to do using a subclass of the _TlsRecordHandler, that way you don't have to worry about the buffering and state